src/acp/rule.test.ts

```diff
@@ -702,3 +783,778 @@ describe("getRequiredRulesOnPolicyAll", () => {
     );
   });
 });
+
+describe("createRule", () => {
+  it("returns a acp:Rule", () => {
+    const myRule = createRule("https://my.pod/rule-resource#rule");
+    expect(
+      myRule.has(
+        DataFactory.quad(
+          DataFactory.namedNode("https://my.pod/rule-resource#rule"),
+          DataFactory.namedNode(RDF_TYPE),
+          DataFactory.namedNode(ACP_RULE)
+        )
+      )
+    ).toBe(true);
+  });
+  it("returns an **empty** rule", () => {
+    const myRule = createRule("https://my.pod/rule-resource#rule");
+    // The rule should only contain a type triple.
+    expect(myRule.size).toEqual(1);
+  });
+});
+
+describe("getRule", () => {
+  it("returns the rule with a matching IRI", () => {
+    const rule = mockRule("https://my.pod/rule-resource#rule");
+    const dataset = setThing(createSolidDataset(), rule);
+    const result = getRule(dataset, "https://my.pod/rule-resource#rule");
+    expect(result).not.toBeNull();
+  });
+
+  it("does not return a Thing with a matching IRI but the wrong type", () => {
+    const notARule = createThing({
+      url: "https://my.pod/rule-resource#not-a-rule",
+    });
+    const dataset = setThing(
+      createSolidDataset(),
+      setUrl(notARule, RDF_TYPE, "http://example.org/ns#NotRuleType")
+    );
+    const result = getRule(dataset, "https://my.pod/rule-resource#not-a-rule");
+    expect(result).toBeNull();
+  });
+
+  it("does not return a rule with a mismatching IRI", () => {
+    const rule = mockRule("https://my.pod/rule-resource#some-rule");
+    const dataset = setThing(createSolidDataset(), rule);
+    const result = getRule(
+      dataset,
+      "https://my.pod/rule-resource#another-rule"
+    );
+    expect(result).toBeNull();
+  });
+});
+
+describe("getAgentForRuleAll", () => {
+  it("returns all the agents a rule applies to by webid", () => {
+    const rule = mockRule("https://my.pod/rule-resource#some-rule", {
+      agents: ["https://my.pod/profile#me", "https://your.pod/profile#you"],
+    });
+    const agents = getAgentForRuleAll(rule);
+    expect(agents).toContainEqual("https://my.pod/profile#me");
+    expect(agents).toContainEqual("https://your.pod/profile#you");
+  });
+
+  it("does not return the groups/public/authenticated a rule applies to", () => {
+    const rule = mockRule("https://my.pod/rule-resource#some-rule", {
+      groups: ["https://my.pod/group#a-group"],
+      public: true,
+      authenticated: true,
+    });
+    const agents = getAgentForRuleAll(rule);
+    expect(agents).not.toContainEqual("https://my.pod/group#a-group");
+    expect(agents).not.toContainEqual(ACP_AUTHENTICATED);
+    expect(agents).not.toContainEqual(ACP_PUBLIC);
+  });
+});
+
+describe("setAgentInRule", () => {
+  it("sets the given agents for the rule", () => {
+    const rule = mockRule("https://my.pod/rule-resource#rule");
+    const result = setAgentInRule(rule, [
+      "https://my.pod/profile#me",
+      "https://your.pod/profile#you",
+    ]);
```

Comment:
Like in #520, I wouldn't make this accept an array.

Reply:
Fixed in b67fa299daec8a7a7fe05d5d50ba370b824e25ad